spec/isec2_linux.go

```diff
@@ -51,3 +55,34 @@ func isEC2() bool {
 
 	return false
 }
+
+func isEC2UUID(uuid string) bool {
+	conds := func(uuid string) bool {
+		if strings.HasPrefix(uuid, "ec2") || strings.HasPrefix(uuid, "EC2") {
+			return true
+		}
+		return false
+	}
+
+	if conds(uuid) {
+		return true
+	}
+
+	// Check as littele endian.
+	// see. https://docs.aws.amazon.com/ja_jp/AWSEC2/latest/UserGuide/identify_ec2_instances.html
+	fields := strings.Split(uuid, "-")
+	src := []byte(fields[0]) // UUID field: time_low(uint32)
```

Comment:
[memo] This is OK, because strings.Split always returns non-empty slice while the separator argument is not empty string.